src/strategies/silo/SiloStrategy.sol

```diff
@@ -243,17 +275,32 @@ contract SiloStrategy is BaseHealthCheck, TradeFactorySwapper {
      * @param . The address that is depositing into the strategy.
      * @return . The available amount the `_owner` can deposit in terms of `asset`
      *
-     
+    */
     function availableDepositLimit(
-        address _owner
+        address // _owner
     ) public view override returns (uint256) {
-        TODO: If desired Implement deposit limit logic and any needed state variables .
+        if (silo.depositPossible(address(asset), address(this))) {
+            ISilo.AssetStorage memory _assetState = silo.assetStorage(address(asset));
 
-        EX:
-            uint256 totalAssets = TokenizedStrategy.totalAssets();
-            return totalAssets >= depositLimit ? 0 : depositLimit - totalAssets;
+            uint256 _price = repository.priceProvidersRepository().getPrice(address(asset));
+            uint256 _totalDepositsValue =
+                _price *
+                (_assetState.totalDeposits + _assetState.collateralOnlyDeposits) /
+                (10 ** ERC20(address(asset)).decimals());
+
+            uint256 _maxDepositsValue = IGuardedLaunch(address(repository)).getMaxSiloDepositsValue(address(silo), address(asset));
+            if (_maxDepositsValue == type(uint256).max) return type(uint256).max;
+
+            if (_maxDepositsValue > _totalDepositsValue) {
+                uint256 _availableDepositValue = _maxDepositsValue - _totalDepositsValue;
+                return _availableDepositValue * _PRECISION / _price;
```

Comment:
this return value will always be in 18 decimals.

`uint256 _price = repository.priceProvidersRepository().getPrice(address(asset));`
`_price` is always in 18 decimals per Silo implementation.
in the end you return the value as:

`_availableDepositValue` (18 decimal, in terms of ETH) * `_PRECISION` (18 decimal) / `_price` (18 decimal)
hence, always 18 decimals. However, it should return the value scaled to our `asset` decimals

This would be the correct way:
`return (_availableDepositValue * 10 ** ERC20(address(asset)).decimals()) / _price`

Reply:
Have you tried this with an asset with 6 decimals like USDC? I think it would've failed to deposit the amounts because of this check

Reply:
Additionally, I would add "-1" to be safe for the rounding errors.

```solidity
uint256 _availableDepositValue = _maxDepositsValue - _totalDepositsValue;
if (_availableDepositValue == 0) return 0;
return ((_availableDepositValue * 10 ** ERC20(address(asset)).decimals()) / _price) - 1;
```

Reply:
Should be fixed at https://github.com/johnnyonline/yVaults-v3-strategies/pull/8/commits/ce40b43ec2dec0e9843874c59fecff630b67aea2. 

I think USDC tests would have caught that one yea. Added now. Looks good.

Reply:
hmm, I think this is also not good enough. If the available deposit is dust then the operation:
`(_availableDepositValue * _decimals / _price)`
can be "0" and when we substract "1" it will underflow.
I think instead of checking this:
`uint256 _availableDepositValue = _maxDepositsValue - _totalDepositsValue;`

check this if its "0" or not and then do the substraction:
`(_availableDepositValue * _decimals / _price)`